### DISCUSSION TITLE
Add developer mode that will not send unlock requests

### DISCUSSION THREAD

**Discussion by redwizard42** [RetroAchievements/RAIntegration#990] [Ideas]
Pretty much every time I need to test an achievement, usually for a ticket, I unlock one or more achievements in softcore mode that I then have to remember to reset.  It would be nice if there were a 'devcore' mode where no unlock requests would be sent to the server, though an 'achievement not unlocked due to...' popup would be necessary still.

Ideally, the developer would be prompted to enter this when opening an appropriate window (achievement editor? assets list? others?), with a "don't ask me this again' checkbox available.  Loading a game with achievements should still prompt to enter hardcore, which would exit this mode if chosen.

Often I'll modify memory and thus invalidate unlocks anyway, but at least 50% of the time I forget to do this, or it just doesn't naturally occur. Having this prompt/mode would be helpful to avoid this situation.

**Comment by redwizard42** [comment]
Also, my expectation for a "don't ask me again" would be that it doesn't prompt OR automatically change modes.  "don't ask me again" isn't a hard requirement in my vision. I gleefully ignore the "do you want to enable hardcore?" prompt all the time when it suits me, afterall,

**Comment by Jamiras** [comment]
I see two relatively simple solutions for this. 

The first is to extend the "Non-Hardcore Warning" to also prompt whether or not to allow softcore unlocks.

> Enable Hardcore mode?
> You are loading a game with achievements and do not currently have hardcore mode enabled.
> Yes/No
 
If "No":
 
> Disable Unlocks?
> Do you also want to prevent unlocking achievements?
> Yes/No

We'd probably also want to show the second prompt when disabling hardcore (either via menu or any of the confirmations from opening developer windows).

The other solution would be to have a new menu item "Disable Softcore Unlocks" that could be permanently enabled by the user (per client), similar to the "Non-Hardcore Warning" menu item. This would be the "don't ask me again" solution.

**Comment by redwizard42** [comment]
This is one of the times I want to say, "Yes. Both please."

Prompt for disabling unlocks if I say no to enabling hardcode / do something that forces hardcore to disabled, plus an option for no softcore ever sounds grand.
